goagen/gen_app/generator.go

```diff
@@ -415,3 +417,81 @@ func (g *Generator) generateUserTypes() error {
 	}
 	return utWr.FormatCode()
 }
+
+// mediaTypeMetadataImports will grab all imports specified in a struct:field:type Metadata tag and
+// add them to the imports for the media_types.go file.
+func (g *Generator) mediaTypeMetadataImports(s []*codegen.ImportSpec) []*codegen.ImportSpec {
```

Comment:
It's to avoid infinite recursion, there can be cycles where an attribute uses another (e.g. an object) which ends up using the first one (e.g. recursive type definition).